Jellyfin.Plugin.AniDB/Providers/equals_check.cs

```diff
@@ -21,527 +19,212 @@ public Equals_check(ILogger<Equals_check> logger)
             {
+                return input;
             }
-            return a;
-        }
 
-        /// <summary>
-        /// Clear name heavy.
-        /// Example: Text & Text to Text and Text
-        /// </summary>
-        /// <param name="a"></param>
-        /// <returns></returns>
-        public async static Task<string> Clear_name_step2(string a, CancellationToken cancellationToken)
-        {
-            if(a.Contains("Gekijyouban"))
-               a= (a.Replace("Gekijyouban", "") + " Movie").Trim();
-            if (a.Contains("gekijyouban"))
-               a = (a.Replace("gekijyouban", "") + " Movie").Trim();
-            try
+            int newLength = (int)((float)input.Length - (((float)input.Length / 100f) * (float)p));
+
+            if (newLength < minLength)
             {
-                a = a.Trim().Replace(await One_line_regex(new Regex(@"(?s) \(.*?\)"), a.Trim(), cancellationToken, 0), "");
+                newLength = minLength;
             }
-            catch (Exception)
-            { }
-            a = a.Replace(".", " ");
-            a = a.Replace("-", " ");
-            a = a.Replace("`", "");
-            a = a.Replace("'", "");
-            a = a.Replace("&", "and");
-            a = a.Replace(":", "");
-            a = a.Replace("␣", "");
-            a = a.Replace("2wei", "zwei");
-            a = a.Replace("3rei", "drei");
-            a = a.Replace("4ier", "vier");
-            return a;
+
+            return input.Substring(0, newLength);
         }
 
         /// <summary>
-        /// If a and b match it return true
+        /// Escape string for regex, but fuzzy
         /// </summary>
         /// <param name="a"></param>
-        /// <param name="b"></param>
         /// <returns></returns>
-        public async static Task<bool> Compare_strings(string a, string b, CancellationToken cancellationToken)
+        public static string FuzzyRegexEscape(string a)
         {
-            if (!string.IsNullOrEmpty(a) && !string.IsNullOrEmpty(b))
-            {
-                if (await Simple_compare(a, b, cancellationToken))
-                    return true;
-                if (await Fast_xml_search(a, b, cancellationToken))
-                    return true;
+            a = Regex.Escape(a);
+
+            // make characters that were escaped fuzzy
+            a = a.Replace(@"\\", ".?");
+            a = a.Replace(@"\*", ".?");
+            a = a.Replace(@"\+", ".?");
+            a = a.Replace(@"\?", ".?");
+            a = a.Replace(@"\|", ".?");
+            a = a.Replace(@"\{", ".?");
+            a = a.Replace(@"\[", ".?");
+            a = a.Replace(@"\(", ".?");
+            a = a.Replace(@"\)", ".?");
+            a = a.Replace(@"\^", ".?");
+            a = a.Replace(@"\$", ".?");
+            a = a.Replace(@"\.", ".?");
+            a = a.Replace(@"\#", ".?");
+
+            // whitespace
+            a = a.Replace(@"\ ", ".?.?.?");
+            a = Regex.Replace(a, @"\s", ".?.?.?");
```

Comment:
curious, any particular reason why any whitespace is equal to 3 `any` character?

Reply:
So you can write the name not completely, or mainly for titles with `: `,  `//` ,  ` & ` or `  - `, which you don't necessarily can or want to have in your folder name.